src/composition.jl

```diff
@@ -18,35 +18,42 @@ Base.size(A::CompositeMap) = (size(A.maps[end], 1), size(A.maps[1], 2))
 Base.isreal(A::CompositeMap) = all(isreal, A.maps) # sufficient but not necessary
 
 # the following rules are sufficient but not necessary
-function LinearAlgebra.issymmetric(A::CompositeMap)
-    N = length(A.maps)
-    if isodd(N)
-        issymmetric(A.maps[div(N+1, 2)]) || return false
-    end
-    for n = 1:div(N, 2)
-        A.maps[n] == transpose(A.maps[N-n+1]) || return false
+const RealOrComplex = Union{Real,Complex}
+LinearAlgebra.issymmetric(A::CompositeMap) = _issymmetric(A.maps)
+_issymmetric(maps::Tuple{}) = true
+_issymmetric(maps::Tuple{<:LinearMap}) = issymmetric(maps[1])
+function _issymmetric(maps::Tuple{Vararg{<:LinearMap}}) # length(maps) >= 2
+    if maps[1] isa UniformScalingMap{<:RealOrComplex}
+        return issymmetric(maps[1]) && _issymmetric(Base.tail(maps))
+    elseif maps[end] isa UniformScalingMap{<:RealOrComplex}
+        return issymmetric(maps[end]) && _issymmetric(Base.front(maps))
+    else
+        return maps[end] == transpose(maps[1]) && _issymmetric(Base.front(Base.tail(maps)))
     end
-    return true
 end
-function LinearAlgebra.ishermitian(A::CompositeMap)
-    N = length(A.maps)
-    if isodd(N)
-        ishermitian(A.maps[div(N+1, 2)]) || return false
-    end
-    for n = 1:div(N, 2)
-        A.maps[n] == adjoint(A.maps[N-n+1]) || return false
+LinearAlgebra.ishermitian(A::CompositeMap) = _ishermitian(A.maps)
+_ishermitian(maps::Tuple{}) = true
+_ishermitian(maps::Tuple{<:LinearMap}) = issymmetric(maps[1])
```

Comment:
Do you really mean `issymmetric`here, or is this an oversight from copy-paste-modify?

Reply:
Very much a typo from copy paste. I've pushed a new version that is more robust against copy paste errors.